### PR TITLE
Add flake8 dependencies to requirements-dev.txt file

### LINKED ISSUE #1074: Add flake8 to the dependencies
This is a minor change to improve the CI file for GitHub actions. 
- Move flake8 dependencies to `requirements-dev.txt`.

### PR DESCRIPTION
**Pull request by dorothykiz1**
closes #1074

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -18,3 +18,8 @@ pytest-faulthandler<2.0; platform_python_implementation == "PyPy"
 pytest-timeout
 feedparser
 python-hglib
+flake8==4.0.1
+pyflakes==2.4.0
+mccabe==0.6.1
+pycodestyle==2.8.0
+
```
Comment (datapythonista) [inline requirements-dev.txt:25]: Not sure what's the reasoning here, but we want just the latest version of `flake8` (the latest at every time, not the latest now), and no other dependencies. If `flake8` needs those other packages, they'll be installed automatically.

Comment (dorothykiz1) [reply requirements-dev.txt]: Thanks @datapythonista

File: requirements-dev.txt [hunk 2]
```diff
@@ -18,3 +18,5 @@ pytest-faulthandler<2.0; platform_python_implementation == "PyPy"
 pytest-timeout
 feedparser
 python-hglib
+flake8
+
```
Comment (datapythonista) [inline requirements-dev.txt:22]: Small thing, but better to not introduce arbitrary blank lines, and keep our changes to exactly what we want.

**Comment by datapythonista** [comment]
> closes [#1074 ]

The squared brackets around the issue will prevent the issue from being automatically closed when this PR is merged. Not sure what are the exact idioms that GitHub accepts, but `Closes #1074` surely work (I think `closes #1074`, `Close #1074`... are also fine). I think GitHub shows you if you hover after that text, whether this is really going to close the issue or not.

**Comment by dorothykiz1** [comment]
Thank you @datapythonista i didn't know the hovering gives you a clue of what `closes` will do.

**Comment by datapythonista** [comment]
Thanks @dorothykiz1

**Pull request merged**